quinn/src/platform/unix.rs

```diff
@@ -296,6 +301,26 @@ fn prepare_msg(
     } else {
         encoder.push(libc::IPPROTO_IPV6, libc::IPV6_TCLASS, ecn);
     }
+
+    if let Some(segment_size) = transmit.segment_size {
+        debug_assert!(
+            U::caps().gso,
+            "Platform must support GSO for setting segment size"
+        );
```

Comment:
I/O errors currently manifest as an error-level tracing event followed by the endpoint driver halting, which can be very hard to diagnose if the application doesn't have tracing wired up. I'm not *super* worked up about being maximally helpful if somebody explicitly mangles their configuration, but arranging things such that `segment_size` is statically guaranteed to be unset if `caps().gso` is false would be more robust.

Reply:
I agree. I noticed that the IO errors are not super visible while implementing the other cmsg stuff.
However I don't see a super good way to have the static guarantees here. The messages are produced by some other code block. And after `poll_transmit` has produced such a message there is not we can do anymore besides failing (in one or the other way).

The main checks that we should look out for is that the max datagram amount which will get passed to `poll_transmit` is only 1 if feature isn't supported. However this should be in another place.

Reply:
> The main checks that we should look out for is that the max datagram amount which will get passed to poll_transmit is only 1 if feature isn't supported

If we'll have a mechanism like that in place, then this error condition is no longer (theoretically) reachable via user misconfiguration, in which case a debug_assert is a perfectly adequate second line of defense.